### PR TITLE
ci: add Thread backend for current stable CRuby

### PR DESCRIPTION
**Pull request by tikkss**
GitHub: GH-235

Because it's an advanced feature not a basic feature.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -83,16 +83,28 @@ jobs:
           - name: "3.3"
             ruby-version: "3.3"
             runs-on: "ubuntu-latest"
+          - name: "3.3 (Thread)"
```
Comment (kou) [inline .github/workflows/test.yml:86]: Could you use `+ XXX` style?

```suggestion
          - name: "3.3 + Thread"
```

Comment (tikkss) [reply .github/workflows/test.yml]: Yes I can. I've switched to the `+ XXX` style.
Thanks for your suggestion!

**Comment by kou** [comment]
Thanks.

**Comment by tikkss** [comment]
Thanks for your review!

**Pull request merged**